app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -341,6 +343,19 @@ private void getCurrentAssignment()
            currentAssignmentSet = new SetCurrentMinistry().doInBackground(this);
         }
     }
+
+    private void refreshCurrentAssignment()
+    {
+        String chosenMinistry = preferences.getString("chosen_ministry", null);
+
+        if(chosenMinistry == null || currentMinistry == null || chosenMinistry.equals(currentMinistry.getName()))
+        {
+            return;
+        }
+
+        refreshAssignment = true;
+        refreshAssignment = !new SetCurrentMinistry().doInBackground(this);
```

Comment:
you are just calling it incorrectly. What you actually want to do for AsyncTasks is something like this: `new SetCurrentMinistry().execute(this);`

then sometime later the method `SetCurrentMinistry.onPostExecute(result)` will be triggered on the UI thread for you to handle what it is returning.